test/test_ffs.cpp

```diff
@@ -74,31 +75,41 @@ TORRENT_TEST(clz)
 		std::fprintf(stderr, "%s\n", t.first);
 		std::uint32_t buf[5];
 		to_binary(t.first, buf);
-		TEST_EQUAL(aux::clz_sw({buf, 5}), t.second);
-		TEST_EQUAL(aux::clz_hw({buf, 5}), t.second);
-		TEST_EQUAL(aux::clz({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros_sw({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros_hw({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros({buf, 5}), t.second);
 	}
 }
 
-TORRENT_TEST(flz_u32)
+TORRENT_TEST(find_last_zero_u32)
 {
 	std::uint32_t v = 0;
-	TEST_EQUAL(aux::flz_sw(v), 31);
-	TEST_EQUAL(aux::flz_hw(v), 31);
-	TEST_EQUAL(aux::flz(v), 31);
+	TEST_EQUAL(aux::find_last_zero_sw(v), 31);
+	TEST_EQUAL(aux::find_last_zero_hw(v), 31);
+	TEST_EQUAL(aux::find_last_zero(v), 31);
 
 	v = 0xffffffff;
-	TEST_EQUAL(aux::flz_sw(v), -1);
-	TEST_EQUAL(aux::flz_hw(v), -1);
-	TEST_EQUAL(aux::flz(v), -1);
-
-	v = 0xff00ff00;
-	TEST_EQUAL(aux::flz_sw(v), 23);
-	TEST_EQUAL(aux::flz_hw(v), 23);
-	TEST_EQUAL(aux::flz(v), 23);
-
-	v = 0xff0fff00;
-	TEST_EQUAL(aux::flz_sw(v), 19);
-	TEST_EQUAL(aux::flz_hw(v), 19);
-	TEST_EQUAL(aux::flz(v), 19);
+	TEST_EQUAL(aux::find_last_zero_sw(v), -1);
+	TEST_EQUAL(aux::find_last_zero_hw(v), -1);
+	TEST_EQUAL(aux::find_last_zero(v), -1);
+
+	v = aux::host_to_network(0xff00ff00);
+	TEST_EQUAL(aux::find_last_zero_sw(v), 31);
+	TEST_EQUAL(aux::find_last_zero_hw(v), 31);
+	TEST_EQUAL(aux::find_last_zero(v), 31);
```

Comment:
This is (and will be) confusing to me too :) I think the hex-literal is written like in our minds it is big-endian, regardless of the host-endianness. Since we are using `htonl` I need to write the literal in the form that we read it in little-endian. Now, the way the bits are indexed is a little odd (from left to right) but it was that way to easily accommodate to the semantic of the new `bitfield` methods. I can document the the bit index order (preferably in another PR)